rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/locations/impl/TreeMapLookup.java

```diff
@@ -101,12 +101,18 @@ private static boolean rangeContains(Range a, Range b) {
 
     @Override
     public @Nullable T lookup(Range from) {
-        T result = contains(data.floorEntry(from), from);
-        if (result == null) {
-            // could be that it's at the start of the entry
-            result = contains(data.ceilingEntry(from), from);
+        // since we allow for overlapping ranges, it might be that we have to
+        // search all the way to the "bottom" of the tree to see if we are
+        // contained in something larger than the closest key
+        var previousKeys = data.headMap(from, true).descendingMap();
+        for (var candidate : previousKeys.entrySet()) {
```

Comment:
I'd rather read the type of `candidate` here instead of `var`. I'm getting lost. Same for `previousKeys`.